intents/time_compare.py

```diff
@@ -0,0 +1,180 @@
+"""
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+"""This module contains the slice-compare intent.
+The slice-compare intent can give the result so that user can easily
+compare the data according to the way user want.
+Also it supports some operations like cropping based on date range,
+slicing(removing rows that do not follow the conditions), group by.
+Some of the operations are optional.
+"""
+
+from util import aspects
+import pandas
+
+def time_compare(table, metric, dimensions, all_dimensions, time_compare_column,
+                                                    summary_operator, **kwargs):
+
+    """ This function returns both the results according to the intent
+    as well as the debiasing suggestions.
+    Some of the oversights considered in this intent are-
+    Args:
+        table: Type-pandas.dataframe
+            It has the contents of the csv file
+        metric: Type-string
+            It is the name of the column according to which grouping will be done.
+            summary operator is applied on metric. Metric could a column
+            containing strings, if we are applying count operator on it.
+        dimensions: Type-list of str
+            It is the name of column we want.
+            'compare batsman A and B according to total_runs',
+             dimension is 'batsman'. we group by dimensions.
+        all_dimension: Type-list of str
```

Comment:
1. Is all_dimensions used anywhere ?  Would it be used in any of the oversights?
2. nit : all_dimension**s**

Reply:
It might be used in top down error.

Reply:
resolved the second part also